### ISSUE TITLE
Cannot import apps with multiple versions

### ISSUE THREAD

**Issue by jvalkeal** [spring-cloud/spring-cloud-dataflow#3287]
Trying to import app uri's from a same file containing multiple versions silently fails. For example:
```
source.time=maven://org.springframework.cloud.stream.app:time-source-rabbit:2.0.1.RELEASE
source.time.metadata=maven://org.springframework.cloud.stream.app:time-source-rabbit:jar:metadata:2.0.1.RELEASE
source.time=maven://org.springframework.cloud.stream.app:time-source-rabbit:2.0.0.RELEASE
source.time.metadata=maven://org.springframework.cloud.stream.app:time-source-rabbit:jar:metadata:2.0.0.RELEASE
sink.log=maven://org.springframework.cloud.stream.app:log-sink-rabbit:2.0.2.RELEASE
sink.log.metadata=maven://org.springframework.cloud.stream.app:log-sink-rabbit:jar:metadata:2.0.2.RELEASE
sink.log=maven://org.springframework.cloud.stream.app:log-sink-rabbit:2.0.1.RELEASE
sink.log.metadata=maven://org.springframework.cloud.stream.app:log-sink-rabbit:jar:metadata:2.0.1.RELEASE
```
Only latest versions per type/name are actually imported. This is most likely because import functionality were created before we really had versioned apps, thus import code probably tracks imports per type/name and latter always overrides previous definitions.

With above example you end up having just `sink.log 2.0.1.RELEASE` and `source.time 2.0.0.RELEASE`

**Comment by jvalkeal** [comment]
This is actually super annoying thing to fix. We need metadata uri versions as well to properly match these during an import. Kinda loosely related to previous fixes in ac5fa963ac2e0eb48173c75768fdc5fa3a0eec52 where we did kinda hacked version check in a dataflow side without requiring deployer spi's to understand their versions.

But essentially for every resource type, (file, maven, docker) we need to be able to extract version for uri to properly differentiate every import from each other together with matching metadata counterpart.

**Comment by jvalkeal** [comment]
When above 8 lines import is processed there should be exactly 4 entries identified by `<type>.<name>.<version>` where each have info about its main uri and metadata uri. Then dataflow internal import feature calculates different apps per version, we just need to send these separately. Also we can just use import resource uri so there's fixes is two different places required.
